Slim/Player/Squeezebox2.pm

```diff
@@ -537,6 +538,21 @@ sub directHeaders {
 
 				$songHandler->handleDirectError($client, $url, $response, $status_line);
 			}
+			elsif ($track->can('redir') && $track->redir && $track->redir ne $url) {
+				
+				# if we have been redirected and we fail on directstream, give it a shot with
+				# the initial utl. This is not a ideal solution, but we have no other option
+				main::INFOLOG && $directlog->is_info && $directlog->info("retrying with non-redirected url ", $track->redir);
+				
+				$controller->song->streamUrl($track->redir);
+				$client->play({
+					'paused'     => ($client->isSynced(1)),
+					'format'     => ($client->master())->streamformat(),
```

Comment:
Are the additional brackets required here?... 

And I don't understand the `paused` definition: how is it related to `isSynced()`?

Reply:
Oh, just saw this is basically a c/p from further down the file. Let's assume that's fine.